### DISCUSSION TITLE
Effect of grain size on calculating orientation density function

### DISCUSSION THREAD

**Discussion by nargesmbg** [mtex-toolbox/mtex#1973] [Ask Anything]
hi everyone, 
I have a question regarding ODF calculation. in which of the following way the grain size are considered in the calculation: 
a) odf{i} = calcDensity(ebsd{i}('Ni').orientations, 'halfwidth',5*degree)

b) o = grains{i}('Ni').meanOrientation
    gs = grains{i}('Ni').diameter
    cs = ebsd{i}('Ni').CS;
    odfWeighted = calcDensity(o,'halfwidth',5*degree,'weights',gs)
    pfWeighted = calcPoleFigure(odfWeighted, h,r)

is it true to say when we are using ebsd.orientation instead of grain.meanorientation the calculated ODF consider grain sizes too?

Best

**Comment by kilir** [comment]
Hi,
in a) you consider each indexed pixel, so it's an pixle area weighted odf.  In b) you weight the odf with the length of the reconstructed grain  - which I think is a little odd. Weighting by the size of the reconstructed grain instead of the actual measurement point makes sense if you have poor indexing, hover the measure should rather be grains.area.

In case you have a material with highly deformed or twinned grains, considering all orientations of all measurement points instead of a single meanorientation might be the better idea.

Hope this helps.

Cheers,
Rüdiger
